test/e2e.minified.js

```diff
@@ -0,0 +1,9 @@
+// Simple check to make sure window.web3 is defined for the karma min loading tests
+describe('window.web3 is defined', function () {
+
+  // Referencing `process` throws w/out browserify.
+  // We want to be skipped during the browserified units tests here.
+  try { if (process) return } catch (e) {}
+
+  if (!window.Web3) throw new Error('"window.Web3" was not defined');
```

Comment:
I think it would be great if we extend this test case with expectations for the values we do request from infura. The ``innerHtml`` property of those should at least not be empty.

Reply:
This test doesn't have an html associated with it. It's only the injection of the test into the karma runner with the min in the headless context. There's now external browserification so if I put a `require('path')` at the top of the file, I would see
```
Uncaught ReferenceError: require is not defined
  at test/e2e.minified.js:1:1
```
That how the test is different from the existing browser tests. Likewise, if I set the preset env to `usage`, I will also see `require is not defined`,  but coming from the bundle.

That's all the test does - it only replicates the crash from #3256.

The E2E requires visual inspection because it happen out on a published website.

Reply:
Oh! Sorry than I was wrong :) 

> The E2E with the html requires visual inspection because it happens out on a published website.

I know from other ``e2e`` test suites that they do compare screenshots for such cases would this be possible or is this too hard to achieve?

Reply:
I think it is possible and it's very common in websites. 

The thing I like about the surge test is that it's simple and real. You make a bundle and publish it to a site. You look at it and it "works". There's no testing framework.

Reply:
I wrote that down in my notes and we can discuss further advanced in-browser tests on the next team call. A small demo ERC-20 DApp as I was creating for my presentation once is probably something we could add as an in-browser e2e test.